web/src/components/RunDetailTest/Visualization.tsx

```diff
@@ -70,26 +70,43 @@ const Visualization = ({runState, spans, type}: IProps) => {
     return <SkeletonDiagram />;
   }
 
-  return type === VisualizationType.Dag ? (
-    <DAG
-      edges={edges}
-      isMatchedMode={matchedSpans.length > 0 || isOpen}
-      matchedSpans={matchedSpans}
-      nodes={nodes}
-      onNavigateToSpan={onNavigateToSpan}
-      onNodesChange={onNodesChange}
-      onNodeClick={onNodeClick}
-      selectedSpan={focusedSpan}
-    />
-  ) : (
-    <Timeline
-      isMatchedMode={matchedSpans.length > 0 || isOpen}
-      matchedSpans={matchedSpans}
-      onNavigateToSpan={onNavigateToSpan}
-      onNodeClick={onNodeClickTimeline}
-      selectedSpan={selectedSpan?.id ?? ''}
-      spans={spans}
-    />
+  return (
+    <>
+      {type === VisualizationType.Dag && (
+        <DAG
+          edges={edges}
+          isMatchedMode={matchedSpans.length > 0 || isOpen}
+          matchedSpans={matchedSpans}
+          nodes={nodes}
+          onNavigateToSpan={onNavigateToSpan}
+          onNodesChange={onNodesChange}
+          onNodeClick={onNodeClick}
+          selectedSpan={focusedSpan}
+        />
+      )}
+
+      {type === VisualizationType.Timeline && (
```

Comment:
In a follow up version, lets consider moving this to a map instead of the ifs

Reply:
to do this, I would need to ensure all Graphs have the same props.
I don't think it's very relevant for this. But I can send a separate PR to deal with this. @xoscar